### ISSUE TITLE
Help compiling and using PowerShell xSCSMA module resources

### ISSUE THREAD

**Issue by ricardogaspar2** [puppetlabs/puppetlabs-dsc#279]
Hey,
I would like to import the DSC resources of this module: https://github.com/PowerShell/xSCSMA .
But the resources' names start with MSFT_x, and I'm afraid of not being able to use them in puppet code with the standard names (e.g. dsc_xSCSMAPowerShellSetup) because of this.

I assume that the renaming of the files and folders will not be enough. Even the MOF files contain the names in this form.

Thanks in advance

**Comment by jpogran** [comment]
There are two names for a DSC Resource: `class name` and `friendly name`. When we build the puppet types, we use the `friendly name`. As example, the DSC Resource you chose: https://github.com/puppetlabs/puppetlabs-dsc/blob/master/lib/puppet/type/dsc_xscsmapowershellsetup.rb which is the `dsc_xscsmapowershellsetup` puppet type.  By convention, bother `friendly name` and `class name` should be the same, but can be different. The situations where they need to be different are very small, and I do not know why the author of this module chose to do it as none of the usual reasons (namespace collision or existing resource using that name...etc) are present. It is more common that they are the same than they are not.

In short, you are safe to use the `friendly name` for all resources you need.

**Comment by ricardogaspar2** [comment]
Ok, thanks! I didn't notice that xSCSMA was already in this repo. Sorry for that.
But  I don't see the ruby types for xRunbookDirectory and xSmaVariable, despite they are inside the puppet_x folder (https://github.com/puppetlabs/puppetlabs-dsc/tree/master/lib/puppet_x/dsc_resources/xSCSMA/DSCResources).

Can I use this resources in puppet manifests?

**Comment by jpogran** [comment]
They are present: https://github.com/puppetlabs/puppetlabs-dsc/blob/master/lib/puppet/type/dsc_smavariable.rb and https://github.com/puppetlabs/puppetlabs-dsc/blob/master/lib/puppet/type/dsc_runbookdirectory.rb

Again, I do not know why the author is mixing and matching the names. For instance `xRunbookDirectory` is the `class name`, but has `RunbookDirectory` as the `friendlyname`

**Comment by jpogran** [comment]
Hi,

Puppet will be shutting down the github issues tab in this repository as there already is a publicly accessible [issue tracker](https://tickets.puppetlabs.com/browse/MODULES) on March 24th, 20017.

If you want to continue the discussion from this issue, please follow the ticket already created from this issue (typically MODULES-XXXX) in our issue tracker.

If you would like to file a new issue, please follow the [template here](https://tickets.puppetlabs.com/CreateIssue.jspa?pid=10707) to create a new ticket.

Thanks,

James Pogran

**Comment by chelnak** [comment]
Hello! We are doing some house keeping and noticed that this issue has been open for a long time.\n\nWe're going to close it but please do raise another issue if the issue still persists. 😄
